### PR TITLE
re-generating error for 'char-map' missing in context.pbtxt

### PR DESCRIPTION
**Pull request by dsindex**
- related issue : https://github.com/tensorflow/models/issues/323#event-763293517
- it is not easy to re-generate same situation. so, i made 'ud_train' directory
- how to generate error
  - remove following lines in `UD/context.pbtxt` and run `./train.sh`
  
  ``` shell
  input {
    name: 'char-map'
    creator: 'brain_pos/greedy'
  }
  ```
  - error message
  
  ``` shell
  + /path/to/models/syntaxnet/bazel-bin/syntaxnet/parser_trainer --task_context=/path/to/models/syntaxnet/ud_train/UD/context.pbtxt --arg_prefix=brain_pos --compute_lexicon --graph_builder=greedy --training_corpus=training-corpus --tuning_corpus=tuning-corpus --output_path=/path/to/models/syntaxnet/ud_train/UD/tmp/syntaxnet-output --batch_size=256 --decay_steps=3600 --hidden_layer_sizes=64 --learning_rate=0.08 --momentum=0.9 --seed=0 --params=64-0.08-3600-0.9-0 --num_epochs=12 --report_every=100 --checkpoint_every=1000 --logtostderr
  INFO:tensorflow:Computing lexicon...
  I syntaxnet/lexicon_builder.cc:134] Term maps collected over 204586 tokens from 12543 documents
  I syntaxnet/term_frequency_map.cc:137] Saved 18731 terms to /path/to/models/syntaxnet/ud_train/UD/tmp/syntaxnet-output/brain_pos/greedy/64-0.08-3600-0.9-0/word-map.
  I syntaxnet/term_frequency_map.cc:137] Saved 15710 terms to /path/to/models/syntaxnet/ud_train/UD/tmp/syntaxnet-output/brain_pos/greedy/64-0.08-3600-0.9-0/lcword-map.
  I syntaxnet/term_frequency_map.cc:137] Saved 50 terms to /path/to/models/syntaxnet/ud_train/UD/tmp/syntaxnet-output/brain_pos/greedy/64-0.08-3600-0.9-0/tag-map.
  I syntaxnet/term_frequency_map.cc:137] Saved 50 terms to /path/to/models/syntaxnet/ud_train/UD/tmp/syntaxnet-output/brain_pos/greedy/64-0.08-3600-0.9-0/category-map.
  I syntaxnet/term_frequency_map.cc:137] Saved 47 terms to /path/to/models/syntaxnet/ud_train/UD/tmp/syntaxnet-output/brain_pos/greedy/64-0.08-3600-0.9-0/label-map.
  F syntaxnet/task_context.cc:140] Check failed: input.part_size() == 1 (0 vs. 1)char-map
  ./train.sh: line 107: 24273       (core dumped) ${BINDIR}/parser_trainer --task_context=${CONTEXT} --arg_prefix=brain_pos --compute_lexicon --graph_builder=greedy --training_corpus=training-corpus --tuning_corpus=tuning-corpus --output_path=${TMP_DIR} --batch_size=${BATCH_SIZE} --decay_steps=3600 --hidden_layer_sizes=${POS_HIDDEN_LAYER_SIZES} --learning_rate=0.08 --momentum=0.9 --seed=0 --params=${POS_PARAMS} --num_epochs=12 --report_every=100 --checkpoint_every=1000 --logtostderr
  ```

### CODE REVIEW

**Comment by dsindex** [comment]
I signed it!

2016-08-27 14:37 GMT+09:00 googlebot <notifications@github.com>:

**Comment by olesyaksyon** [comment]
I signed it!

**Comment by xiamx** [comment]
This pull request helps with my case, can we merge it?

**Comment by drpngx** [comment]
@slavpetrov does that look reasonable?

**Comment by girving** [comment]
@slavpetrov @drpngx Looks like this slipped through the cracks.

**Comment by drpngx** [comment]
OK, it looks like this is a bug in tensorflow rather than an improvement to syntaxnet as I had previously thought, right?

Let me see if I can repro the core dump.

**Comment by drpngx** [comment]
OK, it looks like it might be a problem with the operators in syntaxnet. Can you rebuild in debug mode and print out the stack trace?

**Comment by nealwu** [comment]
Is this something we still want?

**Comment by xiamx** [comment]
Yes please. 👍

**Comment by nealwu** [comment]
@slavpetrov @calberti could take a look?

File: syntaxnet/ud_train/convert.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import sys
+from   optparse import OptionParser
+import time
+
+# --verbose
+VERBOSE = 0
+
+if __name__ == '__main__':
+
+	parser = OptionParser()
```
Comment (drpngx) [inline syntaxnet/ud_train/convert.py]: we're doing two space indentation

File: syntaxnet/ud_train/convert.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+#-*- coding: utf8 -*-
+
+import os
+import sys
+from   optparse import OptionParser
```
Comment (drpngx) [inline syntaxnet/ud_train/convert.py]: remove extra spaces. I am wonder if it's better to use the tf flag parsing instead.

File: syntaxnet/ud_train/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+training tagger and parser
```
Comment (drpngx) [inline syntaxnet/ud_train/README.md]: Capitalize

File: syntaxnet/ud_train/train.sh [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+		--output=beam-parsed-$SET-corpus \
+		--arg_prefix=brain_parser \
+		--graph_builder=structured \
+		--model_path=${TMP_DIR}/brain_parser/structured/${GP_PARAMS}/model
+	done
+}
+
+function copy_model {
+	# needs : category-map  label-map	lcword-map  prefix-table  suffix-table	tag-map  tag-to-category  word-map
+	cp -rf ${TMP_DIR}/brain_parser/structured/${GP_PARAMS}/model ${MODEL_DIR}/parser-params
```
Comment (drpngx) [inline syntaxnet/ud_train/train.sh]: also two space indentation

**Comment by nealwu** [comment]
@bogatyy could you take a look at this PR?

**Comment by gunan** [comment]
Closing due to inactivity.

**Pull request closed without merging**